### PR TITLE
Userview page created

### PR DESCRIPTION
**Pull request by kokilavanisumanas**
Userview page created

### CODE REVIEW

File: app/Http/Controllers/User/DashboardController.php [hunk 1]
```diff
@@ -17,7 +17,7 @@ public function index()
 
     public function gettemplates($id)
     {   
-        $categoryId    = Category::where('name','=',$id)->first();
+        $categoryId    = Category::where('slug','=',$id)->first();
```
Comment (Tamiltuckker) [inline app/Http/Controllers/User/DashboardController.php:22]: Need to space after comma.

File: app/Http/Controllers/User/DashboardController.php [hunk 2]
```diff
@@ -31,4 +31,12 @@ public function getquestions($id)
 
         return view('users.getquestions',compact('quizQuestions','quizTemplateId'));
     }
+
+    public function answerview($id)
```
Comment (Tamiltuckker) [inline app/Http/Controllers/User/DashboardController.php]: change function name as viewquizanswer

File: resources/views/users/getquestions.blade.php [hunk 3]
```diff
@@ -9,21 +9,22 @@
...
                     @csrf
                     @php $options = $quizQuestion->quiz_options @endphp
                     <h4> Q{{ $quizQuestion->id }}.{{ $quizQuestion->question }}</h4>
                     <br>
                     @foreach ($options as $key => $option)
-                        <input type="radio" name={{ "quizanswers[$questionId]"}}  value="{{ $option->id }}"> &nbsp;&nbsp;{{ $option->option }} <br>
-                    @endforeach 
-              @endforeach
+                        <input type="radio" name={{ "quizanswers[$questionId]" }} value="{{ $option->id }}">
+                        &nbsp;&nbsp;{{ $option->option }} <br>
```
Comment (Tamiltuckker) [inline resources/views/users/getquestions.blade.php]: remove and nbsp
need to add whatever  class you want

File: resources/views/users/getquestions.blade.php [hunk 4]
```diff
@@ -9,21 +9,22 @@
                 <p>Check Your Brain</p>
             </div>
             <form method="POST" name="quizanswer" action="{{ route('user.dashboard.store') }}">
-              @foreach ($quizQuestions as $questionKey => $quizQuestion)
-                @php $questionId = $quizQuestion->id @endphp
+                @foreach ($quizQuestions as $questionKey => $quizQuestion)
+                    @php $questionId = $quizQuestion->id @endphp
                     @csrf
                     @php $options = $quizQuestion->quiz_options @endphp
                     <h4> Q{{ $quizQuestion->id }}.{{ $quizQuestion->question }}</h4>
```
Comment (Tamiltuckker) [inline resources/views/users/getquestions.blade.php]: change index value start at 1.

File: resources/views/users/viewanswered.blade.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                <p>Check Your Brain</p>
+            </div>
+
+            @foreach ($quizAnswers as $answerKey => $quizAnswer)
+                @php
+                    $quizQuestion = \App\Models\QuizQuestion::where('id', '=', $quizAnswer->quiz_question_id)->first();
+                    $quizOptions = \App\Models\QuizOption::where('quiz_question_id', '=', $quizAnswer->quiz_question_id)->get();
+                @endphp
+
+                <h4> Q{{ $quizQuestion->id }}.{{ $quizQuestion->question }}</h4>
```
Comment (Tamiltuckker) [inline resources/views/users/viewanswered.blade.php]: Need to change index value

File: resources/views/users/viewanswered.blade.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                    @endphp
+
+                    @if ($quizpoint === null)
+                        <input type="radio" name="" value=""> &nbsp;&nbsp;{{ $quizOption->option }}<br>
+                    @elseif($quizpoint->point === 1)
+                        <span style="background-color:rgb(41, 233, 134);"> <input type="radio" name=""
+                                value=""> &nbsp;&nbsp;{{ $quizOption->option }} </span><br>
+                    @else
+                        <span style="background-color:rgb(233, 41, 41);"> <input type="radio" name=""
+                                value=""> &nbsp;&nbsp;{{ $quizOption->option }} </span><br>
```
Comment (Tamiltuckker) [inline resources/views/users/viewanswered.blade.php]: Optimized theese lines and need to add class. remove inline  style.

File: routes/web.php [hunk 7]
```diff
@@ -48,4 +48,5 @@
     Route::get('/dashboard/questions/{slug}', [UserDashboardController::class, 'getquestions'])->name('dashboard.getquestions');
     Route::post('/dashboard/questions/answer', [QuizController::class, 'store'])->name('dashboard.store');
     Route::resource('profiles', ProfileController::class);
+    Route::get('/dashboard/view/{id}', [UserDashboardController::class, 'answerview'])->name('dashboard.view');
```
Comment (Tamiltuckker) [inline routes/web.php]: change function name

**Review by Tamiltuckker — COMMENTED** [review]
Kindly check

File: resources/views/users/getquestions.blade.php (hunk 3 above)
Comment (kokilavanisumanas) [reply resources/views/users/getquestions.blade.php]: front end Theme changes it will not use it . so now i remove &nbsp. Is this Okay

**Comment by Tamiltuckker** [comment]
done

**Pull request closed without merging**